packages/suite/src/views/wallet/coinmarket/savings/setup/components/FiatAmount/index.tsx

```diff
@@ -43,18 +44,21 @@ interface Props {
     customFiatAmountError?: FieldError;
 }
 
-const FiatAmount = ({
-    control,
+const FiatAmount = <TFieldValues extends Savings>({
     defaultFiatAmount,
     fiatAmount,
     fiatCurrency,
     paymentAmounts,
     minimumPaymentAmountLimit,
     maximumPaymentAmountLimit,
     customFiatAmountError,
-}: Props) => {
+    ...props
+}: FiatAmountProps<TFieldValues>) => {
     const { FiatAmountFormatter } = useFormatters();
 
+    // Type assertion allowing to make the component reusable, see https://stackoverflow.com/a/73624072.
+    const { control } = props as unknown as UseFormReturn<Savings>;
```

Comment:
Do we need to `TFieldValues`? 
<img width="586" alt="Screenshot 2023-06-23 at 10 09 30" src="https://github.com/trezor/trezor-suite/assets/33235762/ca9a4574-2c18-4f6d-8a51-5ca0390c6966">

If I remove `...props` and add `control` to props and remove all the `TFieldValues`, it works the same in my opinion. Looks too complicated now.

Reply:
was it incorrect before or did something change?

Reply:
You are right, it became unnecessary after I refactored the form types and added `Savings` type. Simplified in 2b4edc0868684875c69741b4516def5a121549b2